modular_skyrat/modules/modular_items/lewd_items/code/lewd_arousal/pain.dm

```diff
@@ -19,9 +20,25 @@
 			if(change_amount > 0)
 				adjustArousal(change_amount)
 			if(HAS_TRAIT(src, TRAIT_MASOCHISM))
-				var/pleasure_adjustment = change_amount / 2
-				adjustPleasure(pleasure_adjustment)
+				adjustPleasure(change_amount / 2)
 		pain += change_amount
 	else
 		pain -= abs(change_amount)
-	pain = min(max(pain, 0), 100)
+	pain = clamp(pain, 0, AROUSAL_LIMIT)
+
+// Get damage for pain system
+/datum/species/apply_damage(damage, damagetype, def_zone, blocked, mob/living/carbon/human/affected_mob, forced, spread_damage, wound_bonus, bare_wound_bonus, sharpness, attack_direction)
+	. = ..()
+	if(!.)
+		return
+	if(affected_mob.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+		return
+	var/hit_percent = (100 - (blocked + armor)) / 100
+	hit_percent = (hit_percent * (100 - affected_mob.physiology.damage_resistance)) / 100
+	switch(damagetype)
+		if(BRUTE)
+			var/amount = forced ? damage : damage * hit_percent * brutemod * affected_mob.physiology.brute_mod
+			INVOKE_ASYNC(affected_mob, /mob/living/carbon/human/.proc/adjust_pain, amount)
+		if(BURN)
+			var/amount = forced ? damage : damage * hit_percent * burnmod * affected_mob.physiology.burn_mod
+			INVOKE_ASYNC(affected_mob, /mob/living/carbon/human/.proc/adjust_pain, amount)
```

Comment:
So why exactly is this here and not in the parent proc, as it would technically just be two lines? This feels inefficient like crazy.

Reply:
Good question.